### DISCUSSION TITLE
Upload multiple photos

### DISCUSSION THREAD

**Discussion by shreyas-ananth** [directus/directus#7370] [Archive]
I am trying to upload multiple photos using the directus in the following way by using Typescript. I am trying to make post method for /files endpoint.

```ts
type DmsConfig = {
  url: string;
};
let zeus: any; 
const dmsConfig: DmsConfig = {
  url: import.meta.env.VITE_SD_DMS_URL as string,
};
const initZeusBackend = () => {
  if (dmsConfig) {
    zeus = new Directus(dmsConfig.url);
  } else if (microConfig) {
    zeus = new Directus(microConfig.url);
  }
  return zeus;
};
```

Below I am showing the function for post method: I am passing the file from the onclick event in an upload file button . Here type file: File is imported from a ts file. 

```ts
export type File = {
  document: any;
};

async function uploadAll(file: File) {
  let uploadData = await zeus.files.createOne(file);
  console.log('upload', uploadData);
  return uploadData;
}
```

I am getting the following 400 bad request error : 
Uncaught (in promise) Error: "storage" is required
Please guide me for the following error. I would require the proper example for the post method for /files endpoint in Typescript.

**Comment by rijkvanzanten** [comment]
The JS SDK doesn't currently have a file upload method. See https://docs.directus.io/reference/api/system/files/#upload-a-file for an example on how to upload file contents using JS 👍🏻

**Reply by shreyas-ananth** [reply]
Thanks for your reply !

**Comment by alexvdvalk** [comment]
The SDK supports this. Updated link to docs: https://docs.directus.io/reference/sdk.html#uploading-a-file
